packages/constellation/src/components/Compositions/Input/Input.tsx

```diff
@@ -0,0 +1,73 @@
+import classnames from 'classnames'
+import React, { useState } from 'react'
+
+import { IconVisibilityOff, IconVisibilityOn } from '../..'
+import { InputComponent } from './Input.types'
+
+const Input: InputComponent = ({
+  disabled,
+  helperText = '',
+  id,
+  label = '',
+  password,
+  placeholder,
+  prefix,
+  state,
+}) => {
+  const [showPassword, setShowPassword] = useState(false)
```

Comment:
I like this a lot. I have changed the component to use a type prop instead of the password prop as that is much more versatile. For now I am adding a TODO to add the postfix prop as I also have to add tags when they are done so that will be easier to do at the same time.